aquadoggo/src/libp2p/config.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use libp2p::identity::Keypair;
+use libp2p::swarm::ConnectionLimits;
+use libp2p::Multiaddr;
+use log::info;
+use serde::{Deserialize, Serialize};
+
+use crate::libp2p::identity::Identity;
+
+/// Keypair file name.
+const KEYPAIR_FILE_NAME: &str = "libp2p.pem";
+
+/// QUIC default transport port.
+const QUIC_PORT: u16 = 48648;
+
+/// Libp2p config for the node.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Libp2pConfiguration {
+    /// Dial concurrency factor.
+    ///
+    /// Number of addresses concurrently dialed for a single outbound
+    /// connection attempt.
+    pub dial_concurrency_factor: u8,
+
+    /// Local address.
+    pub listening_multiaddr: Multiaddr,
+
+    /// Maximum incoming connections.
+    pub max_connections_in: u32,
+
+    /// Maximum outgoing connections.
+    pub max_connections_out: u32,
+
+    /// Maximum pending incoming connections.
+    pub max_connections_pending_in: u32,
+
+    /// Maximum pending outgoing connections.
+    pub max_connections_pending_out: u32,
```

Comment:
What is a "pending" connection? Is this some sort of waiting queue?

Reply:
Yes, it's a waiting queue of sorts. A pending connection is one which has been initiated but has not yet received a response.

For example, once we  receive an inbound connection request it is added to the `pending inbound` queue. Once we respond to the request it moves out of the `pending` queue.